modules/using-an-algorithm/pages/index.adoc

```diff
@@ -0,0 +1,110 @@
+The algorithms are divided into several categories.
+Each algorithm has its own folder, where you can find the implementation of the algorithm, any user-defined functions required to install the algorithm, or different versions of the algorithm if they exist.
+
+Once you have decided that an algorithm fits your use case, it's time to run the algorithm on your graph data.
+GDS Library algorithms are implemented in GSQL as GSQL queries.
+Therefore, the process to run an algorithm query is the same as the process to run any other query.
+The names of GDS algorithm queries as provided by the library begin with the prefix `tg_` only to distinguish them from a user-written query.
+However, since the library is open-source, you can change the name of the queries to any name of your choosing.
+
+
+== Check schema/data constraints
+Most algorithm queries in the GDS Library are schema-free, meaning that you are able to run the query on any schema.
+However, some algorithms have certain schema or data constraints by nature.
+Make sure to read the documentation for the algorithm to determine the following:
+
+* Does the algorithm require edges to be directed/undirected?
+* Does the algorithm require edges to be weighted/unweighted?
+* Does the algorithm require any vertex type to have an attribute of a certain data type?
+* Does the algorithm expect your data to have been processed in a certain way before it runs?
+
+For example, xref:classification-algorithms:k-nearest-neighbors.adoc[] runs on graphs with either directed or undirected edges, but the edges must have a weight attribute.
+
+Another example is the xref:node-embeddings:fast-random-projection.adoc[Fast Random Projection algorithm], which expects the vertex type to have an attribute of type `LIST<DOUBLE>` if you want to store the embedding results to your graph data.
+
+== Install necessary UDFs
+
+Some algorithms require the use of user-defined functions (UDF).
+The UDF files are included in the same folder as the algorithm itself.
+
+If your query requires a UDF, ensure that the UDF files are added to the GSQL before creating or installing the query.
+If your query does not require a UDF, you can proceed to create the query.
+
+== Create query
+After making sure your data meets the requirement of the algorithm query, the next step is to create the query in GSQL or save the query in GraphStudio.
+
+
+[tabs]
+====
+GSQL Shell::
++
+--
+. Locate the query in the GDS Library GitHub repository.
+It is a `.gsql` file named after the query.
+. Copy the entire content of the query file.
+This is the command to create the query.
+. Log in to the GSQL shell as a user with query writing privileges for the graph on which you want to create the query.
+. Run `USE GRAPH <graph_name>` to switch to the graph you want to create the query on.
+. Type `BEGIN` to start a multi-line command.
+. Paste the `CREATE QUERY` command into the terminal.
```

Comment:
I would not recommend this method.  I think it is much better to have the query in a file, and then to have GSQL read the file.

Reply:
Updated to paste into file and invoke from GSQL,